src/transformers/tokenization_utils_base.py

```diff
@@ -2365,6 +2451,12 @@ def convert_added_tokens(obj: Union[AddedToken, Any], add_type_field=True):
             legacy_format=legacy_format,
             filename_prefix=filename_prefix,
         )
+        added_files = special_tokens_map_file
+        if self.get_added_vocab():
+            added_files = save_files[-1] + " and " + added_files
+        logger.warning_once(
+            f"Saving {added_files} will be removed in `transformers 5`, it is kept for forward compatibility, but it is recommended to update your tokenizer file"
+        )
```

Comment:
As seen with you, this could be updated to put a warning on load rather than on save